mtp_noms_ops/apps/security/forms/check.py

```diff
@@ -172,3 +172,54 @@ def accept_or_reject(self):
             logger.exception(f'Check {self.object_id} could not be actioned')
             self.add_error(None, _('There was an error with your request.'))
             return False
+
+
+class AssignCheckToUserForm(GARequestErrorReportingMixin, forms.Form):
+    assignment = forms.CharField(max_length=10)
+
+    def __init__(self, object_id, request, **kwargs):
+        super().__init__(**kwargs)
+        self.object_id = object_id
+        self.request = request
+
+    @cached_property
+    def session(self):
+        return get_api_session(self.request)
+
+    def get_object(self):
+        """
+        Gets the security detail object, a sender or prisoner profile
+        :return: dict or None if not found
+        """
+        try:
+            return self.session.get(self.get_endpoint_path()).json()
+        except RequestException:
+            self.add_error(None, _('This service is currently unavailable'))
+            return {}
+
+    def get_endpoint_path(self):
+        return f'/security/checks/{self.object_id}/'
+
+    def assign_or_unassign(self):
+        endpoint = self.get_endpoint_path()
+
+        if self.cleaned_data.get('assignment') == 'assign':
+            user_id = self.request.user.pk
+        elif self.cleaned_data.get('assignment') == 'unassign':
+            user_id = None
+        else:
+            logger.exception(f'Unknown assignment value for check {self.object_id}')
+            self.add_error(None, _('There was an error with your request.'))
+            return False
+        try:
+            self.session.patch(
+                endpoint,
+                json={
+                    'assigned_to': user_id
+                }
+            )
+            return True
+        except RequestException:
+            logger.exception(f'Check {self.object_id} could not be assigned')
+            self.add_error(None, _('There was an error with your request.'))
```

Comment:
similarly, i don't think this will work. `AssignCheckToUserForm.assign_or_unassign` is called from the view only if the form was valid so adding a form error won't allow it show up anywhere. you can, however, use `messages.error(self.request, _('Credit could not be added to your list'))`